virttest/utils_misc.py

```diff
@@ -4558,3 +4558,39 @@ def get_pid(name, session=None):
         return None
     else:
         return int(output.split()[1])
+
+
+def get_sosreport(path=None, remote_ip=None, remote_pwd=None, remote_user="root"):
+    """
+    Get sosreport in host/guest
+
+    :param path: local host path for sosreport to be saved, defaults to logdir
+    :param remote_ip: remote host/guest ip address
+    :param remote_pwd: remote host/guest password
+    :param remote user: remote host/guest username
+
+    :return: sosreport log path on success, None on fail
+    """
+    from avocado.core import data_dir
+    from virttest import remote
+    cmd = "sosreport --batch --all-logs"
```

Comment:
lets check for sos package instal and try proceed for installing package incase not available.

Reply:
correct, I am trying to do it in preprocess to check for sos package and postprocess to collect sosreport based on the config param to enable/disable sosreport